python/ray/dashboard/modules/reporter/reporter_agent.py

```diff
@@ -558,6 +578,64 @@ def decode(b: Union[str, bytes]) -> str:
 
         return gpu_utilizations
 
+    @staticmethod
+    def _parse_nvsmi_output(nvsmi_stdout: str) -> Dict[int, List[ProcessGPUInfo]]:
+        """Parse the output of nvidia-smi pmon -c 1."""
+        process_utilizations = defaultdict(list)
+        lines = nvsmi_stdout.splitlines()
+        table_header = lines.pop(0).lower().split()[1:]
+        # Base on different versions, the header may be different.
+        gpu_id_index = table_header.index("gpu")
+        pid_index = table_header.index("pid")
+        sm_index = table_header.index("sm")
+        mem_index = table_header.index("mem")
+        for line in lines:
+            if line.startswith("#") or not line.strip():
+                continue
+
+            columns = line.split()
+            if len(columns) < max(gpu_id_index, pid_index, sm_index, mem_index) + 1:
+                continue
+
+            gpu_id, pid, sm, mem = (
+                columns[gpu_id_index],
+                columns[pid_index],
+                columns[sm_index],
+                columns[mem_index],
+            )
+            if pid == "-":  # no process on this GPU
+                continue
+            gpu_id = int(gpu_id)
+            pinfo = ProcessGPUInfo(
```

Comment:
can we use a different type here? Since `gpu_memory_usage` is of type `Megabytes`. It's very confusing for it to be a percentage and may introduce tricky bugs later.